routingpy/routers/google.py

```diff
@@ -506,12 +531,11 @@ def matrix(  # noqa: C901
         if transit_routing_preference:
             params["transit_routing_preference"] = transit_routing_preference
 
-        return self.parse_matrix_json(
+        return self._parse_matrix_json(
             self.client._request("/distancematrix/json", get_params=params, dry_run=dry_run)
         )
 
-    @staticmethod
-    def parse_matrix_json(response):
+    def _parse_matrix_json(self, response):
```

Comment:
hm why this change? this is "public" and `staticmethod` on purpose, we use this in other projects IIRC

Reply:
In this case, it prevented other class methods (_parse_legs) from being called. We'll have to figure out how to rewrite this.

Reply:
I did a small rewrite, none of those functions needed access to the instance, they can all be static actually